python/basix/__init__.py

```diff
@@ -9,6 +9,13 @@
 functionality can be used via this Python interface.
 """
 
+import sys
+
+if sys.platform.startswith("win32"):
+    import os
+
+    os.add_dll_directory("D:/a/basix/install/bin")
```

Comment:
Will this always be the correct install path?

Reply:
No - I'm going to wait for @minrk comment on how to deal with this more elegantly. This is 'expected behaviour' on Windows, see e.g. https://docs.python.org/3/library/os.html#os.add_dll_directory

Reply:
I don't know about elegant, but DLLs are generally searched for on $PATH (there is no LD_LIBRARY_PATH for Windows, just PATH).

As a result, the [typical installation directory for DLLs](https://github.com/zeromq/libzmq/blob/aa885c5a154256612108636b0fb22f44ae0e247a/CMakeLists.txt#L1693) on Windows is with

```
RUNTIME_DESTINATION ${CMAKE_INSTALL_BINDIR}
```

So for CI, you can either add the DLL location to %PATH%, or set the install location to somewhere already on %PATH%.

Reply:
More references:

- conda build [docs on Windows shared libraries](https://docs.conda.io/projects/conda-build/en/stable/resources/use-shared-libraries.html#shared-libraries-in-windows), and
- conda-forge [example cmake build script](https://conda-forge.org/docs/maintainer/knowledge_base/#simple-cmake-based-bldbat).

Reply:
I agree, but Python's object loader ignores the rule "DLLs are generally searched for on $PATH", see: https://docs.python.org/3/library/os.html#os.add_dll_directory

Reply:
Right, I forgot about that whole mess. Conda-forge [patches](https://github.com/conda-forge/python-feedstock/blob/46e4991703e49cf75833699d42d6605147d31aa6/recipe/patches/0007-Add-CondaEcosystemModifyDllSearchPath.patch) Python to search in standard locations DLLs can install to. Python doesn't appear to have any of these by default except the system itself and the Extension DLL parent, at least according to the docs.

I guess the choices are:

1. record libbasix.dll location somewhere during Python build with e.g. `configure_file`, so there's a variable to import and call add_dll_dir on
3. get it from an env variable at runtime
3. install libbasix to one of the 'standard' locations, i.e. system directory
4. copy DLLs into the Python package ([delvewheel](https://pypi.org/project/delvewheel/) takes care of this post-build, like `delocate` or `auditwheel` for mac and linux)

If you want `pip install` to work by default without extra args, I think the only option is to copy libbasix into the Python package on Windows, then you only need `add_dll_directory(Path(__file__).parent)`. While delvewheel will do this, if you want users to get it without an extra step, you'll have to do the copy_file yourself.